### PR TITLE
feat: Updating import in MockERC20 to ERC20 instead of Openzeppelin

### LINKED ISSUE #92: feat: blockful custom erc20 as the mock contract
Prosposal: 
- get the blockful erc20 from [this repo](https://github.com/blockful-io/blockful-contracts/blob/main/contracts/token/ERC20/ERC20.sol)
- use the blockful erc20 instead of the openzeppelin in the swaplace repo
- test for bugs and conflicts and propose the fixes as well (if needed)

### PR DESCRIPTION
**Pull request by RafaDSan**
closes #92 

I updated the code in the file `MockERC20.sol` to import the `ERC20.sol` from this [repo](https://github.com/blockful-io/blockful-contracts/blob/main/contracts/token/ERC20/ERC20.sol).

I also added three new files to test the updated MockERC20, using as a base this [repo](https://github.com/blockful-io/swaplace-contracts/tree/main/test).

### CODE REVIEW

**Comment by 0xneves** [comment]
Hey @RafaDSan, thank you for the contribution. I think I left it a little bit confusing in the issue when referring to testing for bugs and issues, I was referring to issues in the Swaplace protocol and not in the ERC20 contract itself. 

The tests you copied already exist and there is no need to have them in the swaplace repository. 
And besides this first misunderstanding, there is another feature I would like to ask: we should not mix the erc20 contract with the Swaplace contracts, avoiding any sort of confusion for the devs by following a more reasonable approach.

Proposal:
- Delete the ERC20 tests.
- Add the ERC20 files inside the mock folder as in `contracts/mock/ERC20`.
- Adapt the file paths
- Test the system using `yarn test`

File: contracts/interfaces/IERC20.sol
Comment (0xneves) [inline contracts/interfaces/IERC20.sol:1]: move to contracts/mock/ERC20/interfaces

File: contracts/interfaces/IERC20Errors.sol
Comment (0xneves) [inline contracts/interfaces/IERC20Errors.sol:1]: move to contracts/mock/ERC20/interfaces

File: contracts/interfaces/IERC20Permit.sol
Comment (0xneves) [inline contracts/interfaces/IERC20Permit.sol:1]: move to contracts/mock/ERC20/interfaces

File: contracts/mock/MockERC20.sol
Comment (0xneves) [inline contracts/mock/MockERC20.sol:1]: Adapt the file paths to the proposed path

File: test/TestERC20Approval.test.ts
Comment (0xneves) [inline test/TestERC20Approval.test.ts:1]: Delete file

File: test/TestERC20Supply.test.ts
Comment (0xneves) [inline test/TestERC20Supply.test.ts:1]: Delete file

File: test/TestERC20Transfer.test.ts
Comment (0xneves) [inline test/TestERC20Transfer.test.ts:1]: Delete file

**Comment by RafaDSan** [comment]
> > Hey @RafaDSan, thank you for the contribution. I think I left it a little bit confusing in the issue when referring to testing for bugs and issues, I was referring to issues in the Swaplace protocol and not in the ERC20 contract itself.
> > The tests you copied already exist and there is no need to have them in the swaplace repository. And besides this first misunderstanding, there is another feature I would like to ask: we should not mix the erc20 contract with the Swaplace contracts, avoiding any sort of confusion for the devs by following a more reasonable approach.
> > Proposal:
> > 
> > * Delete the ERC20 tests.
> > * Add the ERC20 files inside the mock folder as in `contracts/mock/ERC20`.
> > * Adapt the file paths
> > * Test the system using `yarn test`
> 
> Requesting changes on each file. Overall description above.

Thank you very much for clarifying things to me! 

I did as you proposed and:

- [x] Deleted the ERC20 tests
- [x] Added the ERC20 files inside a `contracts/mock/ERC20/interfaces` file
- [x] Adpated the file paths in `MockERC20.sol`
- [x] Tested the contract using `yarn test` and it all worked fine.

**Pull request merged**